### PR TITLE
Eslint Only for JS and VUE files

### LINKED ISSUE #1936: Linting single source of truth
UI Version: 0.X.X

Description:
Currently we are using prettier and eslint, prettier overrides some rules that can't be sa
- prettier is an eslint extension, but it overrides vue:recommended rules
- prettier quick fix hook is not customizable with eslint rules, so when those rules will work we will not be able to commit the fixes

We should decide what set of rules to use and have a single source of truth for the code styling

### PR DESCRIPTION
**Pull request by sabau**
Closes #1936 

**Description:**
- Eslint with plugins to format vue and js
- Added eslint:recommended rules
- removed prettier from the chain and precommit hook

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: app/src/renderer/components/common/TextBlock.vue [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <template>
+  <!-- TODO:SECURITY try to add a malicious block and see how is rendered, like <script>alert('danger')</script> -->
```
Comment (fedekunze) [inline app/src/renderer/components/common/TextBlock.vue]: is there an issue for this ?

File: app/src/renderer/components/common/TmBarDiscrete.vue [hunk 2]
```diff
@@ -25,7 +25,7 @@ export default {
       type: Number,
       default: 0
     },
-    "click-fn": {
+    "clickFn": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmBarDiscrete.vue]:
```suggestion
    clickFn: {
```

File: app/src/renderer/components/common/TmBtn.vue [hunk 3]
```diff
@@ -51,7 +68,7 @@ export default {
       type: String,
       default: null
     },
-    "icon-pos": {
+    "iconPos": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmBtn.vue]:
```suggestion
    iconPos: {
```

File: app/src/renderer/components/common/TmFormGroup.vue [hunk 4]
```diff
@@ -19,15 +21,15 @@ export default {
       type: Boolean,
       default: false
     },
-    "field-id": {
+    "fieldId": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmFormGroup.vue]:
```suggestion
    fieldId: {
```

File: app/src/renderer/components/common/TmFormGroup.vue [hunk 5]
```diff
@@ -19,15 +21,15 @@ export default {
...
       type: String,
       default: null
     },
-    "field-label": {
+    "fieldLabel": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmFormGroup.vue]:
```suggestion
    fieldLabel: {
```

File: app/src/renderer/components/common/TmFormGroup.vue [hunk 6]
```diff
@@ -19,15 +21,15 @@ export default {
...
       type: String,
       default: null
     },
-    "sub-label": {
+    "subLabel": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmFormGroup.vue]:
```suggestion
    subLabel: {
```

File: app/src/renderer/components/common/TmPage.vue [hunk 7]
```diff
@@ -71,7 +73,7 @@ export default {
       type: String,
       default: ``
     },
-    "menu-body": {
+    "menuBody": {
```
Comment (fedekunze) [inline app/src/renderer/components/common/TmPage.vue]:
```suggestion
    menuBody: {
```

File: app/src/renderer/components/common/TextBlock.vue (hunk 1 above)
Comment (sabau) [reply app/src/renderer/components/common/TextBlock.vue]: Not yet but I was thinking of tagging all the possible security holes as TODO:SECURITY, so when we will have an internal security audit some possible weak points are highlighted

File: app/src/renderer/components/common/TmPage.vue (hunk 7 above)
Comment (sabau) [reply app/src/renderer/components/common/TmPage.vue]: I should add a rule for all those, after lunch I'll add this 👍

File: app/src/renderer/components/common/TextBlock.vue (hunk 1 above)
Comment (sabau) [reply app/src/renderer/components/common/TextBlock.vue]: opened 👍

File: tasks/gaia.js [hunk 8]
```diff
@@ -102,7 +102,8 @@ async function makeValidator(
     chainId
   )
   console.log(`Waiting for funds to delegate`)
-  while (true) {
+  const forever = true
+  while (forever) {
```
Comment (faboweb) [inline tasks/gaia.js]: an ignore eslint would maybe have been better

Comment (jbibla) [reply tasks/gaia.js]: lol

**Pull request merged**